cmd/process-agent/main_common.go

```diff
@@ -219,6 +220,14 @@ type miscDeps struct {
 func initMisc(deps miscDeps) error {
 	initRuntimeSettings()
 
+	// Set memory limit
+	go func() {
+		err := runtime.RunMemoryLimiter("process_config", context.TODO())
```

Comment:
Nit: I think typically `context.Background()` is preferred unless there's a future change that is anticipated to replace the `TODO`.
```suggestion
		err := runtime.RunMemoryLimiter("process_config", context.Background())
```